### ISSUE TITLE
Rename `currentFreight` to `lastPromotedFreight`

### ISSUE THREAD

**Issue by alexmt** [akuity/kargo#842]
"Current freight" is misleading since Kargo does not assess what stage is running. We should rename "Current freight" to "last promoted" to resolve the confusion.

**Comment by TheSwarnim** [comment]
Hey @alexmt , I wanted to pick this issue

**Comment by krancour** [comment]
@TheSwarnim I'd like us to hold off on this please...

#631 is underway and it's a pretty significant refactor of the back-end.

I want to wait and do this _after_ that is taken care of in order to avoid some really heinous merge conflicts.

After #631 is all set, this is all yours.

**Comment by TheSwarnim** [comment]
Thanks @krancour, I was going through the source code and set up the local project setup. Meanwhile I'll check if is their any good first issue I can start working on it!

**Comment by jessesuen** [comment]
This issue would just be a rename of the existing field. But agree it should be held off until FreightCRD is implemented

Related to the issue is: https://github.com/akuity/kargo/issues/751

I will leave #751 open as whether or not we need `observedFreight`. IMO, it is not a priority to implement robust, out-of-band drift detection. But at least with the rename of currentFreight to lastPromotedFreight, the implications of what the field means is clearer.

**Comment by rbreeze** [comment]
@krancour Based on our discussion yesterday, it sounds like the `currentFreight` field is being deprecated and ultimately removed, as it's redundant.

Given that, going to close this issue... feel free to reopen if I misunderstood this

(see https://github.com/akuity/kargo/pull/2206)

**Comment by krancour** [comment]
Good catch @rbreeze. This issue definitely is moot now.
